src/operation/union/CascadedPolygonUnion.cpp

```diff
@@ -94,13 +97,58 @@ CascadedPolygonUnion::Union()
     // TODO avoid creating this vector and run binaryUnion off the iterators directly
     std::vector<const geom::Geometry*> geoms(index.items().begin(), index.items().end());
 
-    return binaryUnion(geoms, 0, geoms.size());
+// #ifdef GEOS_OPENMP
+    return pairUnion(geoms);
+// #else
+//     return binaryUnion(geoms, 0, geoms.size());
+// #endif
 }
 
+#ifdef GEOS_OPENMP
+#include <omp.h>
+#endif
 
+/* private */
 std::unique_ptr<geom::Geometry>
-CascadedPolygonUnion::binaryUnion(const std::vector<const geom::Geometry*> & geoms,
-                                  std::size_t start, std::size_t end)
+CascadedPolygonUnion::pairUnion(
+    std::vector<const geom::Geometry*>& inply) const
+{
+    int sz = (int)(inply.size());
```

Comment:
if you used size_t here and in other places, I guess you could avoid the explicit casts below

Reply:
Annoyingly, OpenMP on Windows insists that the loop iterators be `int`, which then cascades into all these wonderful casts. 🤷

Reply:
ok, a comment in the code would help other readers then